botocore/paginate.py

```diff
@@ -623,11 +625,17 @@ def _extract_paging_params(self, kwargs):
             max_items = int(max_items)
         page_size = pagination_config.get('PageSize', None)
         if page_size is not None:
-            if self._pagination_cfg.get('limit_key', None) is None:
+            if self._limit_key is None:
                 raise PaginationError(
                     message="PageSize parameter is not supported for the "
                             "pagination interface for this operation.")
-            page_size = int(page_size)
+            input_members = self._model.input_shape.members
+            limit_key_shape = input_members.get(self._limit_key)
+            if isinstance(limit_key_shape, StringShape):
```

Comment:
I prefer `type_name` over `StringShape` for the following reasons:

1) It is more consistent throughout the botocore codebase. Just grepping through the botocore codebase `type_name` is used considerably more than `StringShape`. When grepping for `StringShape`, I see it only being used a couple of times in the `model` module

2) It is entirely possible for us to change the class name of `StringShape`, but it is way less possible for the `type_name` to ever change. This value is directly pulled from the JSON model and if the literal value `"string"` was to ever change it would break not just our SDK, but probably all of the other SDKs.

3) It seems weird that we would import `StringShape` just for an `isinstance` check when we can just use `type_name`. Relying on an import for instance checks seems to defeat the purpose of having all of the shape classes from the `model` module subclassing from the `Shape` class and having a `type_name` property.